tests/test_deployments/test_app.py

```diff
@@ -269,3 +275,56 @@ def get_missing(messages: List[str]) -> List[str]:
         assert not get_missing(before_stopping), get_logs()
         self.tendermint_node.stop()
         assert not get_missing(after_stopping), get_logs()
+
+
+class TestTendermintBufferFailing(BaseTendermintServerTest):
+    """Test Tendermint buffer"""
+
+    perform_monitoring = False  # Setting this flag to False makes the buffer to fill and freeze Tendermint
+    debug_tendermint = True  # This will cause more logging -> faster failure
+
+    @wait_for_node_to_run
+    def test_tendermint_buffer(self) -> None:
+        """Test Tendermint buffer"""
+
+        try:
+            # Give the test 30 seconds for it to throw a timeout
+            for _ in range(30):
+                # If it hangs for 5 seconds, we assume it's not working.
+                # Increasing the timeout should have no effect,
+                # the node is unresponsive at this point.
+                requests.get("http://localhost:26657/status", timeout=5)
+                time.sleep(1)
+        except requests.exceptions.Timeout:
+            raise AssertionError("Tendermint has timed out")
+
+    @classmethod
+    def teardown_class(cls) -> None:
+        """Teardown the test."""
+        cls.app_context.pop()
+        # Stopping doesn't work when monitoring is off
+        if cls.perform_monitoring:
+            cls.tendermint_node.stop()
```

Comment:
Could you explain this comment?

Reply:
That comes from Ardian. I think he means that when this test fails due to the node timing out, we cannot call stop anymore, so we only call stop when we are monitoring.

Reply:
I had to remove this as tendermint was not being stopped properly and made other tests fail.